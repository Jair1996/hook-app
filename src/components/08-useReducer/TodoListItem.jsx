export const TodoListItem = ({ todo, index, handleToggle, handleDelet }) => {
  return (
    <li className={`list-group-item`}>
      <p
        onClick={() => handleToggle(todo.id)}
        className={`${todo.done && "complete"}`}
      >
        {index + 1}. {todo.desc}
      </p>
      <button onClick={() => handleDelet(todo.id)} className="btn btn-danger">
        Borrar
      </button>
    </li>
  );
};
