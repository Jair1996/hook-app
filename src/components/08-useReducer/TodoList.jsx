import { TodoListItem } from "./TodoListItem";

export const TodoList = ({ todos, handleToggle, handleDelet }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, i) => (
        <TodoListItem
          key={todo.id}
          todo={todo}
          index={i}
          handleToggle={handleToggle}
          handleDelet={handleDelet}
        />
      ))}
    </ul>
  );
};
