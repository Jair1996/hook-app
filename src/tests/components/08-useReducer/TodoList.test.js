import { shallow } from "enzyme";
import { TodoList } from "../../../components/08-useReducer/TodoList";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas en <TodoListItem />", () => {
  const handleToggle = jest.fn();
  const handleDelet = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleToggle={handleToggle}
      handleDelet={handleDelet}
    />
  );

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener dos <TodoListItem />", () => {
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    expect(wrapper.find("TodoListItem").at(0).prop("handleDelet")).toEqual(
      expect.any(Function)
    );
  });
});
