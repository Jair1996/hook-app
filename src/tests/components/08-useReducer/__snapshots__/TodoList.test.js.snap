// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> debe de mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelet={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelet={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender JS",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
